File: backend/app/schemas/backup.py
from datetime import datetime
from pydantic import Field
from app.schemas.common import ORMBase


class BackupInfo(ORMBase):
    nome: str
    tamanho_bytes: int
    tamanho_legivel: str
    criado_em: datetime
    descricao: str | None = None


class CriarBackupPayload(ORMBase):
    descricao: str | None = Field(default=None, max_length=200)


class ResultadoBackup(ORMBase):
    nome: str
    tamanho_bytes: int
    tamanho_legivel: str
    descricao: str | None = None
    mensagem: str


class ResultadoRestauracao(ORMBase):
    mensagem: str
    backup_seguranca_criado: str
    aviso: str

File: frontend/src/services/backup.ts
import { http } from "./http"

export interface BackupInfo {
  nome: string
  tamanho_bytes: number
  tamanho_legivel: string
  criado_em: string
  descricao: string | null
}

export interface ResultadoBackup {
  nome: string
  tamanho_bytes: number
  tamanho_legivel: string
  descricao: string | null
  mensagem: string
}

export interface ResultadoRestauracao {
  mensagem: string
  backup_seguranca_criado: string
  aviso: string
}

export const backupService = {
  listar: () =>
    http.get<BackupInfo[]>("/backup").then(r => r.data),

  criar: (descricao?: string) =>
    http.post<ResultadoBackup>("/backup", { descricao: descricao || null })
      .then(r => r.data),

  downloadUrl: (nome: string) =>
    `/api/v1/backup/${encodeURIComponent(nome)}/download`,

  excluir: (nome: string) =>
    http.delete(`/backup/${encodeURIComponent(nome)}`),

  restaurar: (nome: string) =>
    http.post<ResultadoRestauracao>(`/backup/${encodeURIComponent(nome)}/restaurar`)
      .then(r => r.data),

  upload: (arquivo: File, descricao?: string) => {
    const fd = new FormData()
    fd.append("arquivo", arquivo)
    if (descricao) fd.append("descricao", descricao)
    return http.post<ResultadoBackup>("/backup/upload", fd, {
      headers: { "Content-Type": "multipart/form-data" }
    }).then(r => r.data)
  },
}